container_main.py

```
@@ -66,19 +66,27 @@ def run_scheduler():
+        os.environ["FLASK_ENV"] = "production"
+
+        # Strictly validate PORT to prevent command injection
+        port_str = os.environ.get("PORT", "5000")
+        if not port_str.isdigit():
+            logger.error(f"Invalid PORT value: {port_str}")
+            sys.exit(1)
+
+        # Launch gunicorn using subprocess for proper production deployment
+        # Hardened command invocation: arguments are inlined to avoid injection
+        bind_address = f"0.0.0.0:{port_str}"
+        subprocess.run(
```

Ruff (0.14.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)